tests/unit/framework/run_unit_tests.py

```diff
@@ -19,10 +19,30 @@ def run_command(args):
     return result
 
 
+def cleanup():
+    repo_dir = os.path.join(tests_config.MAIN_DIRECTORY_OF_TESTED_PROJECT)
+    script_path = os.path.dirname(os.path.realpath(__file__))
+    result = run_command(["cd", repo_dir])
+    if result.returncode != 0:
+        raise Exception("Cleanup before testing failed!")
+
+    result = run_command(["git clean", "-dxf"])
```

Comment:
Git clean seems to be too much, e.g. when someone has some changes that are not staged and wants to run unit tests he will get his changes wiped out.

Reply:
Fixed